### PR TITLE
Add propagateUpstreamTrace, deprecate propagateTraceContext

### PR DESCRIPTION
**Pull request by agocs**
### What does this PR do?

I accidentally mis-named the flag I added in #488 . This adds the correct flag and prints a warning when you use a deprecated flag.
 
### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

**Review by estherk15 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ To further configure your plugin, use the following custom parameters in your `s
...
 | `mergeStepFunctionAndLambdaTraces` | When set to `true`, Lambda traces merge with the calling Step Functions trace. Defaults to `false`.                                                                                                                                                                                                                                                                                                                          |
 | `enableStepFunctionsTracing`    | Enable automatic subscription of the Datadog Forwarder to Step Function log groups and Step Functions tracing. If no Step Function log groups are configured, then they are automatically created. Requires setting `forwarderArn`. Defaults to `false`.                                                                                                                                                                  |
+| `propagateTraceContext` | When set to `true`, downstream Stepfunction invocation traces merge with upstream Stepfunction invocations. Defaults to `false`. |
```
Comment (nine5two7) [inline README.md:68]: Should we also add some description for "propagateUpstreamTrace". In addition, based on message below, we want to deprecate the previous two, and use "propagateUpstreamTrace". Once "propagateUpstreamTrace" is enabled, the trace context will be passed down to both "downstream" Lambda and StepFunctions. Is that correct? Dose the CI has the same and consistent setting?

File: src/index.ts [hunk 2]
```diff
@@ -247,7 +247,12 @@ module.exports = class ServerlessPlugin {
 
         if (config.mergeStepFunctionAndLambdaTraces || config.propagateTraceContext) {
           this.serverless.cli.log(
-            `mergeStepFunctionAndLambdaTraces or propagateTraceContext is true, trying to modify Step Functions' definitions to add trace context.`,
+            `mergeStepFunctionAndLambdaTraces and propagateTraceContext will be deprecated. Please use propagateUpstreamTrace instead`,
```
Comment (nine5two7) [inline src/index.ts:250]: So propagateUpstreamTrace = propagate to both Lambda and StepFunctions?

Comment (agocs) [reply src/index.ts:250]: Yes, that is consistent with CI

**Review by nine5two7 — APPROVED** [review]
LGTM

**Pull request merged**